vcn_gateways.tf

```diff
@@ -174,14 +174,23 @@ resource "oci_core_route_table" "service_gw" {
 ###################
 # NAT Gateway (NGW)
 ###################
+resource "oci_core_public_ip" "nat_gateway_public_ip" {
+  count          = var.create_nat_gateway == true && var.nat_gateway_public_ip_id == "RESERVED" ? 1 : 0
```

Comment:
Please use these coding conventions: https://oracle-terraform-modules.github.io/terraform-oci-oke/codingconventions.html

Reply:
@hyder  - I was looking into the oke conventions and I think for this line 178 , the "RESERVED" is in capital letters , I believe the expectation is to have in small letters as "reserved" . Apart from this I dont see any other issue . Please confirm

Reply:
It looks like you are checking whether something the user should be using a reserved ip. In this case, the variable should be of named in the imperative form e.g. `use_reserved_ip`. 

Side question: If you want to use a reserved IP, shouldn't you be already creating this separately instead of creating this here? If you create the reserved IP here and you run destroy, you'll lose the resource. I'm wondering what is then the use case of creating the public IP address here?